packages/multicast_dns/test/lookup_resolver_test.dart

```diff
@@ -59,8 +59,8 @@ void testResult2() {
     final ResourceRecord response2 =
         ip4Result('yyy.local', InternetAddress('2.3.4.5'));
     resolver.handleResponse(<ResourceRecord>[response2, response1]);
-    final IPAddressResourceRecord result1 = await futureResult1.first;
-    final IPAddressResourceRecord result2 = await futureResult2.first;
+    final IPAddressResourceRecord result1 = await (futureResult1.first as Future<IPAddressResourceRecord>);
```

Comment:
Confusing typecast, the test fails with 

`IPAddressResourceRecord extends ResourceRecord.`

`type 'Future<ResourceRecord>' is not a subtype of type 'Future<IPAddressResourceRecord>' in type cast`

How can this be resolved?

Reply:
what about `(await futureResult1.first) as IPAddressResourceRecord`?

Reply:
Nope this isn't working.